src/main/java/hudson/model/ExternalRun.java

```diff
@@ -85,6 +86,18 @@ private void setCharset(String c) { // JENKINS-14107
         charset = c;
     }
 
+    private Result parseResult(String text) {
+        Result result;
+        Scanner sc = new Scanner(text);
+        if (sc.hasNextInt()) {
```

Comment:
Probably better to use `Integer.parseInt` + `catch (NumberFormatException` but OK.